sql/src/main/java/io/crate/jobs/JobContextService.java

```diff
@@ -150,6 +150,23 @@ public long killAll() {
         return numKilled;
     }
 
+    public long killJobs(List<UUID> toKill) {
+        // TODO: when this is exposed some Actions should be stopped like it's done in killAll using the killAllListeners
+        long numKilled = 0L;
+        writeLock.lock();
+        try {
+            for (Map.Entry<UUID, JobExecutionContext> entry : activeContexts.entrySet()) {
```

Comment:
imho it would be better to iterate over `toKill` and then make `.get()` calls on `activeContexts`. Because `toKill.contains` has to iterate over each element in the list on each entry in activeContexts